File: python_django/core/customer/infrastructure/in_memory_customer_repository.py
from typing import List

from core.customer.domain.customer import Customer
from core.customer.domain.customer_id import CustomerId
from core.customer.domain.customer_repository import CustomerRepository


class InMemoryCustomerRepository(CustomerRepository):

    _database = {}

    def find_by(self, customer_id: CustomerId) -> Customer:
        return self._database.get(customer_id.value, None)

    def store(self, customer: Customer) -> None:
        self._database[customer.id.value] = customer

    def find_all(self) -> List[Customer]:
        return list(self._database.values())

File: typescript_nestjs/libs/core/src/customer/infrastructure/respository/InMemoryCustomerRepository.ts
import { CustomerRepository } from '../../domain/CustomerRepository';
import { Customer } from '../../domain/Customer';
import { CustomerId } from '../../domain/CustomerId';

export class InMemoryCustomerRepository implements CustomerRepository {
    private _database = new Map<string, Customer>();

    findById(customerId: CustomerId): Promise<Customer | undefined> {
        const customer = this._database.get(customerId.value);

        return Promise.resolve(customer || null);
    }

    store(customer: Customer): Promise<void> {
        this._database.set(customer.id.value, customer);
        return Promise.resolve();
    }
}
